[core] Allow DD_AGENT_HOST and DD_TRACE_AGENT_PORT env variables

brettlangdon opened the pull request:
To help support desired containerization onboarding improvements we need to support `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT` environment variables for configuration.

These changes are the minimal viable changes needed to support these variables. It isn't the most clean, using something like `get_env()` would be better, but we decided to get this out the door since @Kyle-Verhoog has a task to work on standardizing our `DD_*` variables.

palazzem commented on ddtrace/tracer.py:
```diff
@@ -27,8 +27,8 @@ class Tracer(object):
-    DEFAULT_HOSTNAME = environ.get('DATADOG_TRACE_AGENT_HOSTNAME', 'localhost')
-    DEFAULT_PORT = 8126
+    DEFAULT_HOSTNAME = environ.get('DD_AGENT_HOST', environ.get('DATADOG_TRACE_AGENT_HOSTNAME', 'localhost'))
+    DEFAULT_PORT = int(environ.get('DD_TRACE_AGENT_PORT', 8126))
```
Do we have a test for this code path? Such as initializing a new tracer after setting the env vars without using `ddtrace-run`?

brettlangdon commented:
@palazzem I think if I understood what you were looking for correctly, I updated the tests to also run the same hostname check runner, but without using `ddtrace-run`. I also updated those tests to use a non-default port... just to be sure.
